packages/filesystem/src/electron-browser/file-dialog/electron-file-dialog-service.ts

```diff
@@ -90,6 +90,16 @@ export class ElectronFileDialogService extends DefaultFileDialogService {
         return true;
     }
 
+    protected async canRead(uris: MaybeArray<URI>): Promise<boolean> {
+        for (const uri of Array.isArray(uris) ? uris : [uris]) {
+            if (!(await this.fileService.access(uri, FileAccess.Constants.R_OK))) {
+                this.messageService.error(`Cannot read resource at ${uri.path}.`);
+                return false;
+            }
+        }
```

Comment:
I could imagine that this could impact performance for large arrays of URIs because every `access` call is sequential. We can parallelize this fairly well like this:
```suggestion
        const filePaths = await Promise.all((Array.isArray(uris) ? uris : [uris]).map(uri => {
            return !await this.fileService.access(uri, FileAccess.Constants.R_OK) && uri.path || '';
        }).filter(e => e);
        if (filePaths.length) {
            this.messageService.error(`Cannot read ${filePaths.length} resources: ${filePaths.join(', ')}`);
        }
        return !!filePaths.length;
```